src/core/network/network.cpp

```diff
@@ -353,27 +360,27 @@ NetworkInstance::~NetworkInstance() {
     Finalize();
 }
 
-std::pair<IPv4Address, Errno> GetHostIPv4Address() {
-    std::array<char, 256> name{};
-    if (gethostname(name.data(), static_cast<int>(name.size()) - 1) == SOCKET_ERROR) {
-        return {IPv4Address{}, GetAndLogLastError()};
-    }
+std::optional<IPv4Address> GetHostIPv4Address() {
+    const std::string& selected_network_interface = Settings::values.network_interface.GetValue();
+    const auto network_interfaces = Network::GetAvailableNetworkInterfaces();
+    ASSERT_MSG(network_interfaces.size() > 0,
+               "GetAvailableNetworkInterfaces returned no interfaces");
```

Comment:
Please don't assert here. Assert should only be used for things that should logically never happen.

I run third-party software such as any yuzu build except my own in a network sandbox on Linux. This ensures the software sees 0 network interfaces. That will make me hit this assert.

```suggestion
    if (network_interfaces.size() == 0) {
        LOG_ERROR(Network, "GetAvailableNetworkInterfaces returned no interfaces");
        return {};
    }
```